internal/api/front/perses/src/app/project/dashboard/dashboard-details/dashboard-details.component.ts

```diff
@@ -56,4 +106,28 @@ export class DashboardDetailsComponent implements OnInit {
       },
     );
   }
+
+  private convertDashboardFeeds(dashboardFeeds: DashboardFeedModel[]): void {
+    for (const section of dashboardFeeds) {
```

Comment:
Ouch the complexity of this method is high. Probably we'd like to : 
- first : split it 
- second : move it to a separate place so that we could easily test it

If I understand correctly you have a dashboard and you "feed" it with additional info, so maybe a class named DashboardFeeder that would act like a builder, you initialize it with a dashboard and then you give the "food" with setter and then you call a method "feed" or "generate" we can speak about it if you ant so that I can elaborate more

In my opinion, it would improve the readability and would be easy to test

Reply:
I like the idea. Where should be this new class though ? in the same file or elsewhere ?

Reply:
Based on what @celian-garcia is suggesting you will more than one methods. So the class Can be a bit big. Having it in a dedicated file would help to read the code I think.